tendermint/src/lite/types.rs

```diff
@@ -96,57 +66,152 @@ pub trait Commit {
 /// and trusted validator set is sufficient for a commit to be
 /// accepted going forward.
-/// The default implementation returns true, iff at least a third of the trusted
+pub trait TrustThreshold: Copy + Clone {
+    fn is_enough_power(&self, signed_voting_power: u64, total_voting_power: u64) -> bool;
+}
+
+/// TrustThresholdFraction defines what fraction of the total voting power of a known
+/// and trusted validator set is sufficient for a commit to be
+/// accepted going forward.
+/// The [`Default::default()`] returns true, iff at least a third of the trusted
 /// voting power signed (in other words at least one honest validator signed).
 /// Some clients might require more than +1/3 and can implement their own
-/// TrustLevel which can be passed into all relevant methods.
-pub trait TrustThreshold {
+/// [`TrustThreshold`] which can be passed into all relevant methods.
+#[derive(Copy, Clone)]
+pub struct TrustThresholdFraction {
+    numerator: u64,
+    denominator: u64,
+}
+
+impl TrustThresholdFraction {
+    pub fn new(numerator: u64, denominator: u64) -> Self {
+        // TODO: should we make sure we got a proper fraction here?
+        Self {
+            numerator,
+            denominator,
+        }
+    }
+}
+
+// TODO: should this go in the central place all impls live instead? (currently lite_impl)
+impl TrustThreshold for TrustThresholdFraction {
     fn is_enough_power(&self, signed_voting_power: u64, total_voting_power: u64) -> bool {
-        signed_voting_power * 3 > total_voting_power
+        signed_voting_power * self.denominator > total_voting_power * self.numerator
+    }
+}
+
+impl Default for TrustThresholdFraction {
+    fn default() -> Self {
+        Self::new(1, 3)
     }
 }
 
 /// Requester can be used to request [`SignedHeader`]s and [`ValidatorSet`]s for a
 /// given height, e.g., by talking to a tendermint fullnode through RPC.
-pub trait Requester {
-    // TODO(Liamsi): consider putting this trait and the Store into a separate module / file...
-    type SignedHeader: SignedHeader;
-    type ValidatorSet: ValidatorSet;
-
-    /// Request the signed header at height h.
-    fn signed_header<H>(&self, h: H) -> Result<Self::SignedHeader, Error>
-    where
-        H: Into<Height>;
+pub trait Requester<C, H>
+where
+    C: Commit,
+    H: Header,
+{
+    /// Request the [`SignedHeader`] at height h.
+    fn signed_header(&self, h: Height) -> Result<SignedHeader<C, H>, Error>;
 
     /// Request the validator set at height h.
-    fn validator_set<H>(&self, h: H) -> Result<Self::ValidatorSet, Error>
-    where
-        H: Into<Height>;
+    fn validator_set(&self, h: Height) -> Result<C::ValidatorSet, Error>;
 }
 
 /// This store can be used to store all the headers that have passed basic verification
 /// and that are within the light client's trust period.
-pub trait Store {
-    type TrustedState: TrustedState;
-
+pub trait Store<C, H>
```

Comment:
Do we even need this trait in here anymore? We don't use it in the `lite` now, users can be free to make any kind of store they like ...

Reply:
It's gone 🎉